pkg/types/aws/validation/customendpoint.go

```diff
@@ -0,0 +1,50 @@
+package validation
+
+import (
+	"sort"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/types/aws"
+)
+
+var (
+	supportedServices = map[string]bool{"ec2": true, "elb": true, "iam": true, "route53": true, "s3": true}
+
+	validServices = func() []string {
+		validValues := make([]string, len(supportedServices))
+		i := 0
+		for r := range supportedServices {
+			validValues[i] = r
+			i++
+		}
+		sort.Strings(validValues)
+		return validValues
```

Comment:
I'm not sure I get this. I could be missing something but isn't this just iterating through the map above and returning a []string of all the keys?

Reply:
List of services is returned in the 'value not supported' validation error condition, above implementation is similar to this -
https://github.com/openshift/installer/blob/7a05abd21e86ab75c5fc1b1ae042cc9a2940963b/pkg/types/aws/validation/platform.go#L40-L49